src/main/java/org/broadinstitute/hellbender/utils/iterators/IntervalAlignmentContextIterator.java

```diff
@@ -0,0 +1,109 @@
+package org.broadinstitute.hellbender.utils.iterators;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * For special cases where we want to emit AlignmentContexts regardless of whether we have an overlap with a given interval.
+ *
+ * In other words, this will iterate over each base of each interval and emit an alignment context.  If no reads overlap,
+ *  it will emit an empty alignment context.  Each empty alignment context will be a new instance.
+ */
+public class IntervalAlignmentContextIterator implements Iterable<AlignmentContext>, Iterator<AlignmentContext> {
+    private Iterator<AlignmentContext> alignmentContextIterator;
+    private IntervalLocusIterator intervalLocusIterator;
+    private SimpleInterval currentInterval;
+    private AlignmentContext currentAlignmentContext;
+    private SAMSequenceDictionary dictionary;
+
+    public IntervalAlignmentContextIterator(Iterator<AlignmentContext> alignmentContextIterator, IntervalLocusIterator intervalLocusIterator, SAMSequenceDictionary dictionary) {
+        this.alignmentContextIterator = alignmentContextIterator;
+        this.intervalLocusIterator = intervalLocusIterator;
+        this.dictionary = dictionary;
+
+        // Iterate once.
+        advanceIntervalLocus();
+        advanceAlignmentContext();
+        advanceAlignmentContextToCurrentInterval(this.currentInterval);
```

Comment:
If you are mean to advance the alignmentContext to the currentInterval why bother to call ```advanceAlignmentContext()``` in between? of course need to look into the code to fully understand this but it looks a bit odd.